src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,647 @@
+                //that the next Contact will enter the outer loop.
+                if (isNoLastnames) {
+                    break;
                 }
             }
-        } else {
-            for (String nameString : listNameStrings) {
-                String conNameString = '';
-                Integer conCounter = 0;
-                if (UTIL_Describe.isValidField('Contact', nameString.toLowerCase())) {
-                    for (Contact con : cons) {
-                        conCounter++;
-                        if (con.get(nameString) != null) {
-                            if (conNameString != null
-                                    && conNameString != ''
-                                    && cons.size() > 1
-                                    && conCounter > 1) {
-                                if (conCounter != cons.size()) {
-                                    conNameString += ', ';
-                                } else {
-                                    conNameString += ' ' + Label.defaultNamingConnector + ' ';
-                                }
-                            }
-                            conNameString += con.get(nameString);
-                        }
-                    }
+
+            //Put together the Account name
+            String name = processName(cons, lastName, firstName, salutationSpec, additionalFirstNames,
+                                      additionalSalutations, ns, conSize, i, next);
+
+            if (String.isNotBlank(name)) {
+                finalAccountName.add(name);
+            }
+        }
+
+        return ns.namePrefix + stringAccountName(finalAccountName, ns) + ns.nameSuffix;
+    }
+
+    /*******************************************************************************************************
+    * @description This method is reponsbile for concatenating all additional Salutations, FirstNames, and LastName
+    * into a string to return the final Account name.
+    * @param cons is the list of Contacts in the current trigger context.
+    * @param lastName is the string for the current LastName that is in process.
+    * @param firstName is the string for the current FirstName that is in process.
+    * @param salutationSpec is the string for the current salutationSpec that is in process.
+    * @param additionalFirstNames is the list of string for the additional Contacts' FirtNames.
+    * @param additionalSalutations is the list of string for the additional Contacts' Salutations.
+    * @param ns is the NameSpec class.
+    * @param conSize is an integer that holds the number of Contacts.
+    * @param i is an integer that holds the current iteration.
+    * @param next is an integer that holds the next iteration.
+    * @param cons is the list of Contacts
+    * return String.
+    */
+    private static String processName(List<Contact> cons, String lastName, String firstName, String salutationSpec,
```

Comment:
It doesn't appear that "next" & "conSize" params are used.

Additionally, I think we might be able to get rid of the "cons" and "i" params - it seems like we only need the first contact in the list, so passing in the first contact to this method should be sufficient?

So the revised parameter list might be something like `(Contact con, String lastName, String firstName, String salutationSpec, List<String> additionalFirstNames, List<String> additionalSalutations, NameSpec ns)`